lib/cog/command/pipeline/executor.ex

```diff
@@ -299,78 +304,82 @@ defmodule Cog.Command.Pipeline.Executor do
+  defp output_for(:chat, state, _, _) do
+    output   = state.output
+    template = state.template
+
+    case template do
+      {:common, template_name} when template_name in ["error", "unregistered-user"] ->
+        # No "envelope" for these templates right now
+        Evaluator.evaluate(template_name, output)
+      {:common, template_name} ->
+        Evaluator.evaluate(template_name,
+                           Template.with_envelope(output))
+      template_name ->
+        parser_meta = state.current_plan.parser_meta
+        Evaluator.evaluate(parser_meta.bundle_version_id,
+                           template_name,
+                           Template.with_envelope(output))
+    end
   end
+  defp output_for(:non_chat, state, status, message) do
+    envelope = %{status: status,
+                 pipeline_output: state.output}
+    if message do
+      Map.put(envelope, :message, message)
+    else
+      envelope
+    end
+  end
+  defp output_for(:status_only, _state, status, message) do
+    envelope = %{status: status}
+    if message do
+      Map.put(envelope, :message, message)
+    else
+      envelope
+    end
+  end
+
+  ########################################################################
 
   defp succeed_with_response(state) do
-    respond(state)
+    respond(state, :success)
     {:stop, :shutdown, state}
   end
 
   defp succeed_early_with_response(state) do
     # If our pipeline has "run dry" at any point, we'll send a
-    # response back to all destinations, but only if the pipeline was
-    # initiated from chat.
+    # response back to where the pipeline was initiated from.
     #
-    # If it was initiated from a trigger, though, we don't want to
-    # output to any chat destinations. Otherwise, you'd end up with
-    # "mystery" messages saying "Pipeline succeeded but there was no
-    # output!" without indication of what's going on. If we trigger
-    # something that succeeds without output, we'll just be silent in
-    # chat.
-
-    filtered_destinations = if ChatAdapter.is_chat_provider?(state.request.adapter) do
-      state.destinations
-    else
-      state.destinations
-      |> Enum.reject(&(ChatAdapter.is_chat_provider?(&1.adapter)))
-    end
-
-    # TODO: We're not going through the normal "respond" path here, because
-    # we're explicitly setting a common template... if "respond" gets
-    # the smarts to deal with that, then this can just call
-    # succeed_with_response again... might want to wait until the old
-    # template support is removed to do that.
-    Enum.each(filtered_destinations, fn(d) ->
-      ReplyHelper.send("early-exit", %{}, d.room, d.adapter, state.mq_conn)
-    end)
-
+    # We don't want to send output to other destinations, because
+    # you'd end up with "mystery" messages saying "Pipeline succeeded
+    # but there was no output!" without indication of what's going
+    # on.
+    #
+    # TODO: If the "early-exit" template gets more contextual smarts, we
```

Comment:
Additional information about who executed a pipeline, when, from where, what the pipeline was, which stage it got to before returning no output, etc.; something that would give the members of a chat room enough context to know what was going on if they got redirected to from such a pipeline.